### PR TITLE
Code Review Groups: Add drag and drop tests

### PR DESCRIPTION
**Pull request by bencodeorg**
Adds tests for drag and drop behavior (ie, the function `onDragEnd`) in our `CodeReviewGroupsManager` component. Ideally, we'd mount this component and test that invoking some function resulted in expected changes in the DOM, but Enzyme and React Beautiful DnD [have issues playing nicely with each other](https://github.com/atlassian/react-beautiful-dnd/issues/1756) that we could not figure out.

As a workaround, I test against some key props that we pass to child components (effectively the state of the parent component) -- the list of code review groups we'll render, and the set of students who are currently unassigned to a group.

I considered a couple of other workaround we could test this:

1. Pull all of this logic into a custom hook and test that directly. I [read that testing hooks directly](https://kentcdodds.com/blog/how-to-test-custom-react-hooks) is not recommended (they should be tested in context), so I decided against that.
2. Put functions like `onDragEnd` into pure functions and test them directly -- that seemed like we'd have less test coverage than testing via shallow render and checking state via props passed to children, so I went with that approach.

There is some cleanup work I have started separately to clean up some of the helper functions in `CodeReviewGroupsManager` into a separate utils file, but that work I don't think affects our testing approach here.

## Testing story

Ran these tests manually locally.

## Follow-up work

If this pattern seems reasonable, more test cases to come to cover deleting groups and updating group names.

### CODE REVIEW

**Review by molly-moen — APPROVED** [review]
This approach seems reasonable to me!

**Review by sanchitmalhotra126 — APPROVED** [review]
This definitely seems reasonable given the challenges with the library and enzyme. Appreciate the follow through on getting some testing for this!

File: apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx [hunk 1]
```diff
@@ -152,11 +153,14 @@ const addDroppableIdToGroup = group => {
...
     group.droppableId = DROPPABLE_ID_UNASSIGNED;
   } else {
-    group.droppableId = `${DROPPABLE_ID_PREFIX}${group.id}`;
+    group.droppableId = getAssignedGroupDroppableId(group.id);
   }
   return group;
 };
 
+// Exported for a test
+export const getAssignedGroupDroppableId = id => `${DROPPABLE_ID_PREFIX}${id}`;
```
Comment (maddiedierker) [inline apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx]: readability nit: it's clearer to use string concatenation here than to use a template with string interpolation (`DROPPABLE_ID_PREFIX + id`)

**Review by maddiedierker — APPROVED** [review]
left a tiny readability nit, but the approach looks good here 👍🏻

**Pull request merged**